private.tf

```diff
@@ -0,0 +1,55 @@
+module "private_label" {
+  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.2.2"
+  namespace  = "${var.namespace}"
+  name       = "${var.availability_zone}"
+  attributes = ["private"]
+  stage      = "${var.stage}"
+  delimiter  = "${var.delimiter}"
+  tags       = "${var.tags}"
+}
+
+resource "aws_subnet" "private" {
+  count             = "${var.type == "private" ? length(var.names) : 0}"
+  vpc_id            = "${var.vpc_id}"
+  availability_zone = "${var.availability_zone}"
+  cidr_block        = "${cidrsubnet(var.cidr_block, ceil(log(length(var.names), 2)), count.index)}"
+
+  tags = {
+    "Name"      = "${module.private_label.id}${var.delimiter}${element(var.names, count.index)}"
+    "Stage"     = "${module.private_label.stage}"
+    "Namespace" = "${module.private_label.namespace}"
+  }
+}
+
+resource "aws_route_table" "private" {
+  count  = "${var.type == "private" ? length(var.names) : 0}"
+  vpc_id = "${var.vpc_id}"
+
+  tags = {
+    "Name"      = "${module.private_label.id}${var.delimiter}${element(var.names, count.index)}"
+    "Stage"     = "${module.private_label.stage}"
+    "Namespace" = "${module.private_label.namespace}"
+  }
+}
+
+resource "aws_route" "private" {
+  count                  = "${var.type == "private" ? length(var.names) : 0}"
+  route_table_id         = "${element(aws_route_table.private.*.id, count.index)}"
+  nat_gateway_id         = "${var.ngw_id}"
+  destination_cidr_block = "0.0.0.0/0"
+}
+
+resource "aws_route_table_association" "private" {
+  count          = "${var.type == "private" ? length(var.names) : 0}"
+  subnet_id      = "${element(aws_subnet.private.*.id, count.index)}"
+  route_table_id = "${element(aws_route_table.private.*.id, count.index)}"
+}
```

Comment:
I think you'll get error here if `var.type == "public"` because `element()` doesn't work with empty list.

Reply:
same with `public.tf`

Reply:
There are no errors.
See https://github.com/cloudposse/terraform-aws-named-subnets/pull/3#issuecomment-339398470 and https://github.com/cloudposse/terraform-aws-named-subnets/pull/3#issuecomment-339398771

Reply:
@comeanother, looks like you've used full example form README for tests, try with some minimal example and remove `tfstate` files before start

Reply:
@SweetOps I don't understand.
Could you do the test you want me to do and provide the result?

Reply:
nvm